torchtune/modules/tokenizers/_gpt2.py

```diff
@@ -0,0 +1,217 @@
+
+        # Should have added re.IGNORECASE so BPE merges can happen for capitalized versions of contractions
+        # We do not use external regex library, so this is slightly modified version of the original line
+        self.pat = re.compile(
+            r"""'s|'t|'re|'ve|'m|'ll|'d| ?[^\W\d_]+| ?\d+| ?[^\s\W\d_]+|\s+(?!\S)|\s+"""
+        )
+
+        self.unk_id = unk_id
+        self.bos_id = bos_id
+        self.eos_id = eos_id
+        self.pad_id = pad_id
+
+    @property
+    def vocab_size(self) -> int:
+        return len(self.encoder)
+
+    def bpe(self, token: str) -> str:
+        """
+        Returns pair for the given token.
+
+        Args:
+            token (str): Passed token.
+
+        Returns:
+            Pair token for the given token.
+        """
+        if token in self.cache:
+            return self.cache[token]
+        word = tuple(token)
+        pairs = get_pairs(word)
+
+        if not pairs:
+            return token
+
+        while True:
+            bigram = min(pairs, key=lambda pair: self.bpe_ranks.get(pair, float("inf")))
+            if bigram not in self.bpe_ranks:
+                break
+            first, second = bigram
+            new_word = []
+            i = 0
+            while i < len(word):
+                try:
+                    j = word.index(first, i)
+                except ValueError:
+                    new_word.extend(word[i:])
+                    break
+                else:
+                    new_word.extend(word[i:j])
+                    i = j
+
+                if word[i] == first and i < len(word) - 1 and word[i + 1] == second:
+                    new_word.append(first + second)
+                    i += 2
+                else:
+                    new_word.append(word[i])
+                    i += 1
+            new_word = tuple(new_word)
+            word = new_word
+            if len(word) == 1:
+                break
+            else:
+                pairs = get_pairs(word)
+        word = " ".join(word)
+        self.cache[token] = word
+        return word
+
+    def _tokenize(
+        self,
+        text: str,
+    ) -> List[int]:
+        """
+        Tokenize, but not encode given text.
+
+        Args:
+            text (str): text to tokenize
+
+        Returns:
+            BPE Tokens.
+        """
+        bpe_tokens = []
+        for token in re.findall(self.pat, text):
+            token = "".join(
+                self.byte_encoder[b] for b in token.encode("utf-8")
+            )  # Maps all our bytes to unicode strings, avoiding control tokens of the BPE (spaces in our case)
+            bpe_tokens.extend(bpe_token for bpe_token in self.bpe(token).split(" "))
+
+        return bpe_tokens
+
+    def _convert_token_to_id(self, token: str):
+        return self.encoder.get(token, self.encoder.get(self.unk_id))
+
+    def encode(
+        self,
+        text: str,
+        add_bos: bool = True,
+        add_eos: bool = True,
+    ):
```

Comment:
```suggestion
    ) -> List[int]:
```
Need to also update the docstring below. I think once the pre-commit hooks have been reinstall it should catch this.

Reply:
Hey Salman! Thanks for connecting to the review! Yes, it is a reasonable fix